backend/src/articles_service/articles_parser.py

```diff
@@ -0,0 +1,52 @@
+import re
+import os
+from pdfminer.high_level import extract_text
+
+
+def format_text(text, leave_paragraphs=True):
+    text = re.sub("^.{1,2}$", "", text, flags=re.MULTILINE)
+    if leave_paragraphs:
+        text = re.sub("^\n$", "", text, flags=re.MULTILINE)
+        return text
+    text = re.sub("\n$", "", text, flags=re.MULTILINE)
+    return text
+
+
+class ArticleParser:
+    def __init__(self, pdf_path):
+        self.pdf_path = pdf_path
+        self.text = extract_text(self.pdf_path, maxpages=2)
+
+    def get_abstract(self) -> str:
+        abstract = re.findall("(?i)abstract((?:.|\n)(?:.+\n)+)", self.text)
+        if len(abstract) > 0:
+            return format_text(abstract[0], False)
+        raise ValueError("couldn't find the abstract")
+
+    def get_keywords(self) -> list[str]:
+        keywords = re.findall("(?i)keywords:((?:.|\n)(?:.+\n)+)", self.text)
+        if len(keywords) > 0:
+            keywords[0] = keywords[0].strip()
+            keywords = keywords[0].split(", ")
+            return keywords
+        raise ValueError("couldn't find the keywords")
```

Comment:
Please create a separate Exception like ArticleParsingError, this is because we need to differentiate them from other ValueErrors, and messages should start from upper case `Couldn't...`

Reply:
There should be three exceptions all inheriting from ArticleParsingError, 
- AbstractParsingError
- KeywordParsingError
- AuthorsParsingError